blocks/bricks/recurrent/base.py

```diff
@@ -46,6 +46,9 @@ def initial_states(self, batch_size, *args, **kwargs):
             The keyword arguments of the application call.
 
         """
+        if not hasattr(self, 'apply') or not self.apply.states:
+            return
+
```

Comment:
You're right about the `apply` method. The `Brick` subclass should follow some design rules. But no code checks whether there is an `apply` method in a `Brick` subclass at present.